### PR TITLE
"Help" window menu

### PR DESCRIPTION
**Pull request by PavelLaptev**
image

---

@Byron thank you for this package [open-rs](https://github.com/Byron/open-rs) It's great because it works without any fuss 🙂

### CODE REVIEW

**Comment by PavelLaptev** [comment]
@Qix- I'm wondering if we can add the app version there. It could solve this issue https://github.com/gitbutlerapp/gitbutler/issues/3590

if I do this:
```rs
  help_menu = help_menu.add_item(disabled_menu_item(
        "help/version",
        format!("Version {}", _package_info.version).as_str(),
    ));
```

I see this:
image

Is this correct way to display the app version

**Comment by Qix-** [comment]
That might actually work, but you might need to run the tauri bundling step instead of just local dev.

File: crates/gitbutler-tauri/src/menu.rs [hunk 1]
```diff
@@ -1,7 +1,10 @@
+// Importing required libraries and modules
+
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs]: Can you remove this comment?

File: crates/gitbutler-tauri/src/menu.rs [hunk 2]
```diff
@@ -12,28 +15,39 @@ use tracing::instrument;
 
 use crate::error::Error;
 
+// Command to set menu item enabled or disabled
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs:18]: This one too?

File: crates/gitbutler-tauri/src/menu.rs [hunk 3]
```diff
@@ -12,28 +15,39 @@ use tracing::instrument;
...
 #[tauri::command(async)]
 #[instrument(skip(handle), err(Debug))]
 pub async fn menu_item_set_enabled(
     handle: AppHandle,
     menu_item_id: &str,
     enabled: bool,
 ) -> Result<(), Error> {
+    // Retrieve the main window from the app handle
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs:26]: And this one? They aren't adding a whole lot.

File: crates/gitbutler-tauri/src/menu.rs [hunk 4]
```diff
@@ -97,74 +124,132 @@ pub fn build(_package_info: &PackageInfo) -> Menu {
...
+    help_menu = help_menu.add_item(CustomMenuItem::new("help/report-issue", "Report an Issue…"));
+    help_menu = help_menu.add_native_item(MenuItem::Separator);
+    help_menu = help_menu.add_item(CustomMenuItem::new("help/discord", "Discord"));
+    help_menu = help_menu.add_item(CustomMenuItem::new("help/youtube", "YouTube"));
+    help_menu = help_menu.add_item(CustomMenuItem::new("help/x", "X"));
+    help_menu = help_menu.add_native_item(MenuItem::Separator);
+    // add version info
+    help_menu = help_menu.add_item(disabled_menu_item(
+        "help/version",
+        format!("Version {}", _package_info.version).as_str(),
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs]:
```suggestion
        &format!("Version {}", _package_info.version),
```
This should work.

**Review by Qix- — CHANGES_REQUESTED** [review]

File: crates/gitbutler-tauri/src/menu.rs [hunk 5]
```diff
@@ -97,74 +124,132 @@ pub fn build(_package_info: &PackageInfo) -> Menu {
...
+    // DEVELOPER TOOLS MENU ITEM
     #[cfg(any(debug_assertions, feature = "devtools"))]
     {
         if event.menu_item_id() == "view/devtools" {
             event.window().open_devtools();
         }
     }
+
+    // HELP MENU ITEMS
+    if event.menu_item_id() == "help/documentation" {
```
Comment (Qix-) [inline crates/gitbutler-tauri/src/menu.rs]: These `if` statements can be a single match statement.

```rust
let result = match event.menu_item_id() {
    "help/documentation" => open::that("..."),
    "something/else" => open::that("..."),
    unknown => panic!("unknown event emitted"),
};

if let Err(err) = result {
    tracing::error!(error = ?err, "failed to open url for {}", event.menu_item_id());
}
```

Something like that.

**Review by Qix- — APPROVED** [review]

**Pull request merged**